src/test/java/co/refiere/resources/RedeemCodeResourceTest.java

```diff
@@ -2,59 +2,78 @@
-         CompanyDatabase companyData = new CompanyDatabase();
-         companyData.setName("testCompany");
-         CompanyDatabaseDao companyDataBase = new CompanyDatabaseDao();
-         companyDataBase.save(companyData);
-         databaseId = companyData.getId();
-         Assert.assertNotNull(databaseId);
-         
-         Campaign campaign = new Campaign();
-         campaign.setName("test campaig name");
-         campaign.setCompanyDatabase(companyData);
-         CampaingDao campaingDao = new CampaingDao();
-         campaingDao.save(campaign);
-     }
-    
-     
-      @Test
-      public void testGetIt() {
-        NewUserRequest newUserRequest = new NewUserRequest();
-        newUserRequest.setCampaignId(databaseId); 
-        newUserRequest.setReferalPersonId(2);
-        newUserRequest.setIdentificationCardNumber("1274563495");
-        newUserRequest.setName("User Test Name");
-        newUserRequest.setLastName("Last-name");
-        newUserRequest.setEmail("email@test.com");
-        newUserRequest.setPhoneNumber("89898989");
-        Response response = target().path("v1/redeemCode/registerUser").request().post(Entity.json(newUserRequest));
-        assertEquals(200, response.getStatus());
-      }
-}
+
+	private int databaseId;
+	Person referalPerson = new Person();
+	CampaignDao campaignDao = new CampaignDao();
+	Campaign testCampaign;
+	ReferencesCodesDao referencesCodesDao = new ReferencesCodesDao();
+
+
+	@Override
+	protected Application configure() {
+		return new ResourceConfig(RedeemCodeResource.class);
+	}
+
+	@Before
+	public void createTestData(){
+		CompanyDatabase companyData = new CompanyDatabase();
+		companyData.setName("testCompany");
+		CompanyDatabaseDao companyDataBase = new CompanyDatabaseDao();
+		companyDataBase.save(companyData);
+		databaseId = companyData.getId();
+		Assert.assertNotNull(databaseId);
+
+		testCampaign = new Campaign();
+		testCampaign.setName("test campaig name");
+		testCampaign.setCompanyDatabase(companyData);
+		campaignDao.save(testCampaign);
+
+		referalPerson.setIdentificationCardNumber("00000");
+		referalPerson.setName("ReferalName");
+		referalPerson.setLastName("ReferalLastName");
+		referalPerson.setPhoneNumber("0000000");
+		referalPerson.setEmail("referalEmail@test.com");
+		PersonDao personDao = new PersonDao();
+		personDao.save(referalPerson);
+	}
+
+	@Test
+	public void createNewUserTest() {
+		NewUserRequest newUserRequest = new NewUserRequest();
+		newUserRequest.setCampaignId(testCampaign.getId()); 
+		newUserRequest.setReferalPersonId(referalPerson.getId());
+		newUserRequest.setIdentificationCardNumber("1111111");
+		newUserRequest.setName("newUserName");
+		newUserRequest.setLastName("newUserLastName");
+		newUserRequest.setEmail("newUserEmail@test.com");
+		newUserRequest.setPhoneNumber("00000000");
+		Response response = target().path("v1/redeemCode/registerUser").request().post(Entity.json(newUserRequest));
+		assertEquals(200, response.getStatus());
```

Comment:
Check status 200 does not guarantee the user was created or not, cause even though the service creates or not a new user, the end-point is going to return a OK response. Please be more strict in the assert.